compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -509,6 +517,64 @@ encodeModuleRule =
         let (hash, bs) = SS.encodeModule lfVersion m
         return ([], Just (mconcat $ hash : map fst encodedDeps, bs))
 
+-- hlint
+
+-- We need to provide hlint the path to the directory of a file
+-- containing a base configuration file called 'hlint.yaml'.
+-- We test with one version of 'hlint.yaml' and deploy with another
+-- (at the current time, there are good reasons for this). The code
```

Comment:
If there are good reasons, mention them in the comment!

Reply:
I started to but it was getting wordy so I gave up :) Explained in responses in the ticket. We either change this or I have another go.